src/clib/lib/enkf/enkf_obs.cpp

```diff
@@ -419,6 +419,12 @@ static void handle_history_observation(enkf_obs_type *enkf_obs,
             enkf_conf, "HISTORY_OBSERVATION");
     int num_hist_obs = stringlist_get_size(hist_obs_keys);
 
+    if (num_hist_obs > 0 && enkf_obs->refcase == NULL) {
+        stringlist_free(hist_obs_keys);
+        throw exc::invalid_argument(
```

Comment:
Looks good, but probably need to do some cleanup on the caller side as well to free `conf_instance_type`